webapp/src/components/Estate/Estate.container.js

```diff
@@ -1,24 +1,29 @@
 import { connect } from 'react-redux'
-import { locations } from 'locations'
 import { push } from 'react-router-redux'
+
+import { locations } from 'locations'
 import { fetchEstateRequest } from 'modules/estates/actions'
 import { getData as getEstates, getLoading } from 'modules/estates/selectors'
+import { getInitialEstate } from 'shared/estate'
 
 import Estate from './Estate'
 
-const mapState = (state, { id }) => {
+const mapState = (state, { assetId, x, y }) => {
   const estates = getEstates(state)
-  const isLoading = getLoading(state).some(estate => estate.id === id)
-  const estate = estates[id]
+  const estate = assetId ? estates[assetId] : getInitialEstate(x, y)
+  const isLoading = assetId
+    ? getLoading(state).some(estate => estate.asset_id === assetId)
+    : false
+
   return {
     isLoading,
     estate
   }
 }
 
-const mapDispatch = (dispatch, { id }) => ({
-  onLoaded: () => dispatch(fetchEstateRequest(id)),
-  onAccessDenied: () => dispatch(push(locations.estateDetail(id)))
+const mapDispatch = (dispatch, { assetId }) => ({
+  onLoaded: () => assetId && dispatch(fetchEstateRequest(assetId)),
+  onAccessDenied: () => dispatch(push(locations.estateDetail(assetId)))
```

Comment:
That is because I want one component which provides me with an estate (`initialEstate` or  `Estate`) based on the params. In this component its support getting an estate from coordinate if it is new and also by `assetId`